### PR TITLE
Renames from API review

### LINKED ISSUE #2412: Bedrock API review
## Protocol abstractions renamed

Microsoft.AspNetCore.Connections.Abstractions

```C#
public abstract ConnectionHandler
{
    Task OnConnectionAsync(ConnectionContext connectionContext);
}
```

### PR DESCRIPTION
**Pull request by davidfowl**
- Rename Microsoft.AspNetCore.Protocols.Abstractions to Microsoft.AspNetCore.Connections.Abstractions.
- Renamed IConnectionHandler to IConnectionDispatcher (and related properties and types)
- Added ConnectionHandler and UseConnectionHandler extension method to Connections.Abstractions.

Fixes #2412 

/cc @natemcmaster @ryanbrandenburg @pranavkm for rename visibility. I also need to rename packages in SignalR.

### CODE REVIEW

**Comment by davidfowl** [comment]
@halter73 Travis keeps failing because of the Systemd socket activation:

```
$ if test "$TRAVIS_OS_NAME" != "osx"; then bash test/SystemdActivation/docker.sh; fi
test/SystemdActivation/docker.sh: line 7: dotnet: command not found
The command "if test "$TRAVIS_OS_NAME" != "osx"; then bash test/SystemdActivation/docker.sh; fi" exited with 127.
```

**Comment by davidfowl** [comment]
Figured out this needs to be ported to the release branch https://github.com/aspnet/KestrelHttpServer/commit/b90766d09755c663e2af868380006213af1171f4.

**Comment by ryanbrandenburg** [comment]
Thanks for the rename heads up, I'll keep an eye out and let you know if there's any problems.

File: src/Connections.Abstractions/ConnectionBuilderExtensions.cs [hunk 1]
```diff
@@ -4,10 +4,23 @@
...
 using System.Threading.Tasks;
 
-namespace Microsoft.AspNetCore.Protocols
+namespace Microsoft.AspNetCore.Connections
 {
     public static class ConnectionBuilderExtensions
     {
+        public static IConnectionBuilder UseConnectionHandler<TConnectionHandler>(this IConnectionBuilder connectionBuilder) where TConnectionHandler : ConnectionHandler
+        {
+            // REVIEW: We should use ActivatorUtilities here
```
Comment (anurse) [inline src/Connections.Abstractions/ConnectionBuilderExtensions.cs]: Sooo, are you going to then?

Comment (davidfowl) [reply src/Connections.Abstractions/ConnectionBuilderExtensions.cs]: Yah, I'll do it in this PR.

**Comment by halter73** [comment]
@davidfowl @pranavkm Can we merge https://github.com/aspnet/KestrelHttpServer/commit/b90766d09755c663e2af868380006213af1171f4 into release/2.1 to get travis passing on PRs like this?

**Pull request merged**